src/libAtomVM/term.h

```diff
@@ -1106,6 +1111,33 @@ static inline term term_create_empty_binary(size_t size, Heap *heap, GlobalConte
     return t;
 }
 
+static inline bool term_normalize_binary_pos_len(term binary, avm_int_t pos, avm_int_t len, BinaryPosLen *pos_len)
+{
+    avm_int_t size = (avm_int_t) term_binary_size(binary);
+    if (len < 0) {
+        pos += len;
+        len = -len;
+    }
+
+    if (UNLIKELY((pos < 0) || (pos > size) || (pos + len > size))) {
+        return false;
+    }
+
+    pos_len->pos = pos;
+    pos_len->len = len;
+    return true;
+}
+
+static inline bool term_is_invalid_binary_pos_len(BinaryPosLen pos_len)
+{
+    return pos_len.pos == -1 && pos_len.len == -1;
```

Comment:
Let's use pointers in order to avoid memory copies: `BinaryPosLen pos_len` -> `BinaryPosLen *pos_len`.

Reply:
Also, is (pos_len.pos == -1) || (pos_len.len == -1);` a valid one?